api/circulation.py

```diff
@@ -413,27 +415,32 @@ def fulfill(self, patron, pin, licensepool, delivery_mechanism, sync_on_failure=
         return fulfillment
 
     def fulfill_open_access(self, licensepool, delivery_mechanism):
-        # Keep track of a default way to fulfill this loan in case the
-        # patron's desired delivery mechanism isn't available.
+        """Fulfill an open-access LicensePool through the requested
+        DeliveryMechanism.
+
+        :param licensepool: The title to be fulfilled.
+        :param requested_lpdm: A DeliveryMechanism.
+        """
+        if isinstance(delivery_mechanism, LicensePoolDeliveryMechanism):
+            self.log.warn("LicensePoolDeliveryMechanism passed into fulfill_open_access, should be DeliveryMechanism.")
+            delivery_mechanism = delivery_mechanism.delivery_mechanism
         fulfillment = None
         for lpdm in licensepool.delivery_mechanisms:
             if not (lpdm.resource and lpdm.resource.representation
                     and lpdm.resource.representation.url):
-                # We don't actually know how to deliver this
-                # allegedly open-access book.
+                # This LicensePoolDeliveryMechanism can't actually
+                # be used for fulfillment.
                 continue
             if lpdm.delivery_mechanism == delivery_mechanism:
                 # We found it! This is how the patron wants
                 # the book to be delivered.
                 fulfillment = lpdm
                 break
-            elif not fulfillment:
-                # This will do in a pinch.
-                fulfillment = lpdm
```

Comment:
It will not do in a pinch. Either we fulfill the book in the format the patron asked for or we tell them we can't fulfill the book at all. Sending a random format just crashes the client.